test/integration/test_raw_api.py

```diff
@@ -550,6 +551,71 @@ def raw_api_test_helper(raw_api, should_cleanup_old_buckets):
             is_file_lock_enabled=False,
         )
 
+    # b2_100_continue
+    print('b2_100_continue')
+    file_name = 'test-100-continue.txt'
+    file_contents = b'hello world'
+    file_sha1 = hex_sha1_of_stream(io.BytesIO(file_contents), len(file_contents))
+
+    def set_100_header(fn):
+        def wrapper(url, headers, *args, **kwargs):
+            headers.update({
+                'Expect': '100-continue',
+            })
```

Comment:
wait, what, we are not setting this header by default during upload etc? 
I wonder what is the purpose of adding this in if we are not using it

please consult this with @pawelpolewicz  if this should be enabled by default or turned off completely for now

I'm leaving out the "optional" option since IMO having it is for "greater good" (infrastructure wise ;) ) and having it always on may only negatively affect very small file uploads which are not the main use case of b2 AFAIK.